### PR TITLE
added a section for unstaging a staged file.

### PR DESCRIPTION
**Pull request by hasanein**
I have accidentally staged many files in my working copy, and was looking for a way to unstage them by filename as I had some staged files that I wanted to keep. Didn't see anything that addressed my specific issue on the README, so thought of adding it here.

### CODE REVIEW

**Comment by hasanein** [comment]
I have accidentally staged many files in my working copy, and was looking for a way to unstage them by filename as I had some staged files that I wanted to keep. Didn't see anything that addressed my specific issue on the README, so thought of adding it here.

File: README.md [hunk 1]
```diff
@@ -480,6 +481,18 @@ When you want to get rid of all of your untracked files
 $ git clean -f
 ```
 
+<a href="I-want-to-unstage-specific-staged-file"></a>
+### I want to unstage specific staged file
```
Comment (RichardLitt) [inline README.md]: unstage a specific

File: README.md [hunk 2]
```diff
@@ -480,6 +481,18 @@ When you want to get rid of all of your untracked files
...
+
+Sometimes we have one or more files that accidentally ended up being staged, and these files have not been committed before, to unstage them
```
Comment (RichardLitt) [inline README.md]: before. To unstage them:

File: README.md [hunk 3]
```diff
@@ -480,6 +481,18 @@ When you want to get rid of all of your untracked files
...
+
+```sh
+$ git reset -- filename
```
Comment (RichardLitt) [inline README.md]: Put filename in `<brackets>`.

File: README.md [hunk 4]
```diff
@@ -480,6 +481,18 @@ When you want to get rid of all of your untracked files
...
+```
+
+this result in unstaging the file and make it look like it's untracked, then you can decide what to do with it, either delete or add a .gitignore entry.
```
Comment (RichardLitt) [inline README.md]: "This unstages the file, and makes it look like it's untracked."

No need for the second sentence. Sometimes we just want a file to be unstaged.

File: README.md [hunk 5]
```diff
@@ -100,14 +101,14 @@ All commands should work for at least git version 2.13.0. See the [git website](
...
-    - [Push a branch and tag that have the same name](#push-a-branch-and-a-tag-that-have-the-same-name)
+    - [Push a branch and a tag that have the same name](#push-a-branch-and-a-tag-that-have-the-same-name)
   - [Tracking Files](#tracking-files)
     - [I want to change a file name's capitalization, without changing the contents of the file](#i-want-to-change-a-file-names-capitalization-without-changing-the-contents-of-the-file)
     - [I want to overwrite local files when doing a git pull](#i-want-to-overwrite-local-files-when-doing-a-git-pull)
     - [I want to remove a file from Git but keep the file](#i-want-to-remove-a-file-from-git-but-keep-the-file)
     - [I want to revert a file to a specific revision](#i-want-to-revert-a-file-to-a-specific-revision)
     - [I want to list changes of a specific file between commits or branches](#i-want-to-list-changes-of-a-specific-file-between-commits-or-branches)
-    - [I want Git to ignore changes to a file without deleting it](#i-want-git-to-ignore-changes-to-a-specific-file)
+    - [I want Git to ignore changes to a specific file](#i-want-git-to-ignore-changes-to-a-specific-file)
```
Comment (RichardLitt) [inline README.md]: Was this supposed to happen? Did you use the automated doctoc?

File: README.md (hunk 1 above)
Comment (hasanein) [reply README.md]: Nicely spotted. Changed.

File: README.md (hunk 2 above)
Comment (hasanein) [reply README.md]: Thanks. Changed.

File: README.md (hunk 3 above)
Comment (hasanein) [reply README.md]: Done.

File: README.md (hunk 4 above)
Comment (hasanein) [reply README.md]: Removed the 2nd sentence.

File: README.md (hunk 5 above)
Comment (hasanein) [reply README.md]: This seemed to have happened after I ran the doctoc after I added my initial changes. I thought the current branch might had an outdated TOC anyways.

Comment (hasanein) [reply README.md]: Just re-ran the doctoc against master without any of my changes and the following two lines from TOC seem to be outdated:

```
-    - [Push a branch and tag that have the same name](#push-a-branch-and-a-tag-that-have-the-same-name)
+    - [Push a branch and a tag that have the same name](#push-a-branch-and-a-tag-that-have-the-same-name)

-    - [I want Git to ignore changes to a file without deleting it](#i-want-git-to-ignore-changes-to-a-specific-file)
+    - [I want Git to ignore changes to a specific file](#i-want-git-to-ignore-changes-to-a-specific-file)
```

File: README.md [hunk 6]
```diff
@@ -480,6 +481,18 @@ When you want to get rid of all of your untracked files
...
+<a href="I-want-to-unstage-specific-staged-file"></a>
+### I want to unstage a specific staged file
+
+Sometimes we have one or more files that accidentally ended up being staged, and these files have not been committed before. To unstage them:
+
+```sh
+$ git reset -- <filename>
+```
+
+this result in unstaging the file and make it look like it's untracked.
```
Comment (RichardLitt) [inline README.md]: This results in...

File: README.md [hunk 7]
```diff
@@ -480,6 +481,18 @@ When you want to get rid of all of your untracked files
...
+
+
```
Comment (RichardLitt) [inline README.md]: One space only please.

File: README.md (hunk 6 above)
Comment (hasanein) [reply README.md]: Done.

File: README.md (hunk 7 above)
Comment (hasanein) [reply README.md]: Done

File: README.md (hunk 5 above)
Comment (RichardLitt) [reply README.md]: I really don't mind. All good. :)

File: README.md [hunk 8]
```diff
@@ -480,6 +481,17 @@ When you want to get rid of all of your untracked files
...
+<a href="I-want-to-unstage-specific-staged-file"></a>
+### I want to unstage a specific staged file
+
+Sometimes we have one or more files that accidentally ended up being staged, and these files have not been committed before. To unstage them:
+
+```sh
+$ git reset -- <filename>
+```
+
+this results in unstaging the file and make it look like it's untracked.
```
Comment (RichardLitt) [inline README.md]: Capitalize `this`.

Comment (hasanein) [reply README.md]: Done 👍

**Comment by RichardLitt** [comment]
Thank you! Sorry about all of the little points of contention. Good work.

**Pull request merged**